### ISSUE TITLE
Verify that buffalo runs before tests

### ISSUE THREAD

**Issue by adyach** [gomods/athens#452]
There was a change in #449 regarding running buffalo before running any tests. It has to be validated, because buffalo should run all necessary things.

https://github.com/gomods/athens/pull/449/files/722d75649f2a8070e76ec23e3e6e177cb74fc963#diff-f785e9e46b9afd2567df7a658338db17

**Comment by robjloranger** [comment]
I can confirm that I needed to do the following to get unit tests to run and pass.
```console
# docker running
make alldeps

buffalo db create
buffalo db migrate up

env GO_BINARY_PATH=/path/to/go1.11beta3 make test-unit
```

This assumes previously installing other deps, either `make setup-dev-env` or by other means.

This raises another concern, `make setup-dev-env` calls the target for `dev` which only spins up a few services, this should spin up all services via `alldeps` as unit tests require all.

The make target that runs `buffalo test` is `test`, this also requires `alldeps`. It does **not** require creation and migration of the database prior.

One more important conclusion is that the make target `dev` does not seem to be needed. Nor does `down` it is the same as `dev-teardown`.

That got a little off topic, I will move my concerns into a second issue.

---
TL;DR you do need to create and migrate the db to run unit tests.

**Comment by marpio** [comment]
@robjloranger I think we can close this after the recent changes.
